addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/discovery/DiscoveryService.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.yeelight.internal.discovery;
+
+import com.yeelight.sdk.device.DeviceBase;
+import com.yeelight.sdk.listeners.DeviceListener;
+import com.yeelight.sdk.services.DeviceManager;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.yeelight.YeelightBindingConstants;
+import org.openhab.binding.yeelight.handler.YeelightHandlerBase;
+import org.openhab.binding.yeelight.internal.YeelightHandlerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author Coaster Li (lixin@yeelink.net) - Initial contribution
+ */
+public class DiscoveryService extends AbstractDiscoveryService implements DeviceListener {
+
+    private Logger mLogger = LoggerFactory.getLogger(DiscoveryService.class.getSimpleName());
+
+    public DiscoveryService() {
+        super(YeelightHandlerFactory.SUPPORTED_THING_TYPES_UIDS, 2, true);
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        mLogger.debug("StartBackgroundDiscovery");
+        DeviceManager.getInstance().startDiscovery();
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        mLogger.debug("StopBackgroundDiscovery");
+        DeviceManager.getInstance().stopDiscovery();
+    }
+
+    @Override
+    protected void startScan() {
+        mLogger.debug( "Starting Scan");
+        DeviceManager.getInstance().registerDeviceListener(this);
+        DeviceManager.getInstance().startDiscovery();
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        mLogger.debug(": stopScan");
+        DeviceManager.getInstance().stopDiscovery();
+        DeviceManager.getInstance().unregisterDeviceListener(this);
+    }
+
+    @Override
+    public void onDeviceFound(DeviceBase device) {
+        mLogger.info("onDeviceFound, id: {}" , device.getDeviceId());
+        ThingUID thingUID = getThingUID(device);
+
+        if( thingUID == null ) {
```

Comment:
Please auto-format the whole binding using eclipse (with ESH / openHAB settings) to get the standard formatting